huaweicloud/services/sdrs/resource_huaweicloud_sdrs_protected_instance.go

```diff
@@ -0,0 +1,234 @@
+	region := cfg.GetRegion(d)
+	client, err := cfg.SdrsV1Client(region)
+	if err != nil {
+		return diag.Errorf("error creating SDRS client: %s", err)
+	}
+
+	n, err := protectedinstances.Get(client, d.Id()).Extract()
+	if err != nil {
+		if errCode, ok := err.(golangsdk.ErrDefault400); ok {
+			if resp, pErr := common.ParseErrorMsg(errCode.Body); pErr == nil && resp.ErrorCode == "SDRS.0208" {
+				// `SDRS.0208` means invalid protected instance ID
+				return common.CheckDeletedDiag(d, golangsdk.ErrDefault404{},
+					"error retrieving SDRS protected instance")
+			}
+		}
+		return diag.FromErr(err)
+	}
+
+	mErr := multierror.Append(
+		nil,
+		d.Set("region", region),
+		d.Set("name", n.Name),
+		d.Set("group_id", n.GroupID),
+		d.Set("server_id", n.SourceServer),
+		d.Set("description", n.Description),
+		d.Set("target_server", n.TargetServer),
+	)
+	if mErr.ErrorOrNil() != nil {
+		return diag.Errorf("error setting resource: %s", mErr)
+	}
+
+	if err := utils.SetResourceTagsToState(d, client, "protected-instances", d.Id()); err != nil {
+		return diag.FromErr(err)
+	}
+	return nil
+}
+
+func resourceProtectedInstanceUpdate(ctx context.Context, d *schema.ResourceData,
+	meta interface{}) diag.Diagnostics {
+	cfg := meta.(*config.Config)
+	region := cfg.GetRegion(d)
+	client, err := cfg.SdrsV1Client(region)
+	if err != nil {
+		return diag.Errorf("error creating SDRS client: %s", err)
+	}
+
+	if d.HasChange("name") {
+		updateOpts := protectedinstances.UpdateOpts{
+			Name: d.Get("name").(string),
+		}
+		_, err = protectedinstances.Update(client, d.Id(), updateOpts).Extract()
+		if err != nil {
+			return diag.Errorf("error updating SDRS protected instance name, %s", err)
+		}
+	}
+
+	if d.HasChange("tags") {
+		if err := utils.UpdateResourceTags(client, d, "protected-instances", d.Id()); err != nil {
+			return diag.Errorf("error updating tags of SDRS protected instance %s: %s", d.Id(), err)
+		}
+	}
+	return resourceProtectedInstanceRead(ctx, d, meta)
+}
+
+func resourceProtectedInstanceDelete(_ context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	cfg := meta.(*config.Config)
+	region := cfg.GetRegion(d)
+	client, err := cfg.SdrsV1Client(region)
+	if err != nil {
+		return diag.Errorf("error creating SDRS client: %s", err)
+	}
+	// use for WaitForJobSuccess
+	client.Endpoint = client.ResourceBase
+
+	deleteOpts := protectedinstances.DeleteOpts{
+		DeleteTargetServer: d.Get("delete_target_server").(bool),
+		DeleteTargetEip:    d.Get("delete_target_eip").(bool),
+	}
+	n, err := protectedinstances.Delete(client, d.Id(), deleteOpts).ExtractJobResponse()
+	if err != nil {
+		if errCode, ok := err.(golangsdk.ErrDefault400); ok {
+			if resp, pErr := common.ParseErrorMsg(errCode.Body); pErr == nil && resp.ErrorCode == "SDRS.0208" {
+				// `SDRS.0208` means invalid protected instance ID
+				return common.CheckDeletedDiag(d, golangsdk.ErrDefault404{},
+					"error deleting SDRS protected instance")
```

Comment:
```suggestion
				return diag.Errorf("invalid protected instance ID: %s", d.Id())
```

Looks like it doesn't return `SDRS.0208` on delete

Reply:
If `SDRS.0208` is returned when deleting, what should be done to delete it?